src/cudadecoder/thread-pool-light.h

```diff
@@ -99,7 +100,7 @@ class ThreadPoolLightWorker {
         (curr_task_.func_ptr)(curr_task_.obj_ptr, curr_task_.arg1,
                               curr_task_.arg2);
       } else {
-        usleep(1000);  // TODO
+        Sleep(1e-3f);  // TODO
```

Comment:
just imo, this does not really help readability
maybe either rename the function to something like "SleepSeconds" or add defines so that it says Sleep (1000 *kMiliseconds) or something else :)
y

Reply:
Or both rename the function and add the defines/constants
y.

Reply:
I thought about that. There is a much better solution. C++11 (indirectly) support units of measure through user-defined literals. The indirect part is it's up to the implementor to make sure that, for example, mass added to time fails to compile (by treating different units as isomorphic but distinct fields, e.g. the `duration` type that does not add to either unitless double value or and instance of the `mass`, but allows to multiply them, so they commute categorically over `duration×mass`). A full support for arbitrary units¹ would probably consist of a lot of C++17 variadic template trickery, but I vaguely remember Boost had a stab at it.

Anyhow, C++14 stdlib++ includes [duration literals constructors](https://en.cppreference.com/w/cpp/chrono/operator%22%22ms) that support duration units; you could be able to write `Sleep(250ms)`, or pass `const auto kDelay = 250ms;` to it, etc.

When I have a breather, I am going to add an overload to `Sleep()` that would accept the `duration` type, and the rest is already provided by the choro library (essentially, the same implementation as I added minus the first line, although durations are be tricky when converted between different clocks—they are intrinsically defined by clocks). Maybe we're able to deprecate the non-dimensional version some day.

I do not think there is so much ambiguity, as the second is the SI unit, and, besides, most of other absolute durations in Kaldi are in seconds by default (such as frame offset etc.). Anyway, it's inconsistent, and migrating to dimensioned duration may make the code more robust and easier to use. But that's when I have a breather and take care of at least the worst of another 18 issues assigned to me :)

---

¹ My favorite unit of measure (volume, to be specific) that I ever came across is the _buttload_. The definition shifted over time, place, and the exact liquid volume of which was being measured, but an oft cited conversion equals 1 buttload of wine, and sometimes beer, to 4 barrels, so that's certainly quite a _buttload_ of booze, in the modern sense of the word.